// ngx-highlight-js/index.js
import { NgModule } from '@angular/core';
import { HighlightJsDirective } from './components/highlight-js.directive';
import * as ɵngcc0 from '@angular/core';
export { HighlightJsDirective } from './components/highlight-js.directive';
var HighlightJsModule = (function () {
    function HighlightJsModule() {
    }
    /** @nocollapse */
    HighlightJsModule.ctorParameters = function () { return []; };
HighlightJsModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: HighlightJsModule });
HighlightJsModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function HighlightJsModule_Factory(t) { return new (t || HighlightJsModule)(); } });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(HighlightJsModule, { declarations: function () { return [HighlightJsDirective]; }, exports: function () { return [HighlightJsDirective]; } }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(HighlightJsModule, [{
        type: NgModule,
        args: [{
                declarations: [HighlightJsDirective],
                exports: [HighlightJsDirective]
            }]
    }], function () { return []; }, null); })();
    return HighlightJsModule;
}());
export { HighlightJsModule };
