// ngx-highlight-js/components/highlight-js.directive.js
import { Directive, ElementRef, Input } from "@angular/core";
import * as ɵngcc0 from '@angular/core';
var HighlightJsDirective = (function () {
    function HighlightJsDirective(elementRef) {
        this.elementRef = elementRef;
    }
    HighlightJsDirective.prototype.ngOnInit = function () {
        this.codeEl = document.createElement('pre');
        if (this.lang) {
            this.codeEl.className = this.lang;
        }
        this.codeEl.innerHTML = '' + this.elementRef.nativeElement.innerHTML.trim();
        this.parentEl = this.elementRef.nativeElement.parentNode;
        this.parentEl.insertBefore(this.codeEl, this.elementRef.nativeElement.nextSibling);
        hljs.configure(Object.assign({}, this.options));
        hljs.highlightBlock(this.codeEl);
    };
    HighlightJsDirective.prototype.ngOnDestroy = function () {
        if (this.codeEl) {
            this.parentEl.removeChild(this.codeEl);
            this.codeEl = null;
        }
    };
    /** @nocollapse */
    HighlightJsDirective.ctorParameters = function () { return [
        { type: ElementRef, },
    ]; };
    HighlightJsDirective.propDecorators = {
        "options": [{ type: Input },],
        "lang": [{ type: Input },],
    };
HighlightJsDirective.ɵfac = function HighlightJsDirective_Factory(t) { return new (t || HighlightJsDirective)(ɵngcc0.ɵɵdirectiveInject(ɵngcc0.ElementRef)); };
HighlightJsDirective.ɵdir = ɵngcc0.ɵɵdefineDirective({ type: HighlightJsDirective, selectors: [["", "highlight-js", ""]], hostAttrs: [2, "display", "none"], inputs: { options: "options", lang: "lang" } });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(HighlightJsDirective, [{
        type: Directive,
        args: [{
                selector: '[highlight-js]',
                host: {
                    'style': 'display:none;'
                }
            }]
    }], function () { return [{ type: ɵngcc0.ElementRef }]; }, { options: [{
            type: Input
        }], lang: [{
            type: Input
        }] }); })();
    return HighlightJsDirective;
}());
export { HighlightJsDirective };
